eng/pipelines/linux.yml

```diff
@@ -23,65 +23,46 @@ jobs:
       displayName: Linux
       strategy:
         matrix:
-          x64_Release:
-            _BuildConfig: Release
-            _architecture: x64
-            _framework: netcoreapp
-            _helixQueues: $(linuxDefaultQueues)
-            _dockerContainer: rhel7_container
-            _buildScriptPrefix: ''
-            _buildExtraArguments: ''
-            _publishTests: true
-
-          arm64_Release:
-            _BuildConfig: Release
-            _architecture: arm64
-            _framework: netcoreapp
-            _helixQueues: $(linuxArm64Queues)
-            _dockerContainer: ubuntu_1604_arm64_cross_container
-            _buildScriptPrefix: 'ROOTFS_DIR=/crossrootfs/arm64 '
-            _buildExtraArguments: -warnAsError false
-            _publishTests: true
-
-          musl_x64_Release:
-             _BuildConfig: Release
-             _architecture: x64
-             _framework: netcoreapp
-             _helixQueues: $(alpineQueues)
-             _dockerContainer: alpine_36_container
-             _buildScriptPrefix: ''
-             _buildExtraArguments: /p:RuntimeOS=linux-musl
-             _publishTests: true
-
-          ${{ if or(eq(parameters.testScope, 'outerloop'), eq(parameters.testScope, 'all')) }}:
-            x64_Debug:
-              _BuildConfig: Debug
+          ${{ if eq(parameters.fullMatrix, 'false') }}:
+            x64_Release:
+              _BuildConfig: Release
```

Comment:
It looks like this plus:

```yaml
          ${{ if eq(parameters.fullMatrix, 'true') }}:
            x64_Debug:
              _BuildConfig: Debug
```

is causing the official Linux x64 artifacts to be built Debug mode rather than Release mode:

https://dev.azure.com/dnceng/internal/_build/results?buildId=223624 (before)
https://dev.azure.com/dnceng/internal/_build/results?buildId=223845 (after)

I was taking a look at this because the uptake of CoreFX into Core-Setup started failing from the first to the second. Debug mode definitely seems wrong to me, although I don't understand how it would be related to the break. (https://github.com/dotnet/core-setup/pull/6817)

Reply:
> Debug mode definitely seems wrong to me, although I don't understand how it would be related to the break

We don't embed the IBC optimization data in Debug builds so this very likely is the reason for the break. Thanks for root causing this! It wouldn't have occurred to me that this could be an official build configuration issue. (I started looking at product build on my local machine...)